pkg/app/piped/imageprovider/ecr/ecr.go

```diff
@@ -0,0 +1,183 @@
+// Copyright 2020 The PipeCD Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ecr
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ecr"
+	"go.uber.org/zap"
+
+	"github.com/pipe-cd/pipe/pkg/model"
+)
+
+// The maximum number of image results returned by the APIs about images.
+// The API allows this value to be between 1 and 1000.
+// See more: https://pkg.go.dev/github.com/aws/aws-sdk-go/service/ecr#ListImagesInput
+const maxResults = 1000
```

Comment:
If the returned list was already sorted, we can decrease this number.
Not sure but I think they are already sorted so can you make a real request to check their order?

Reply:
you are right, actually i did. as far as i tried, they seem to be sorted by older (means the first value is always the oldest).

Reply:
anyway, seems to be needed to gain all tags. (i know this isn't the best, but i decided to make this a future task because it would require some hack as long as any official API wasn't provided.

Reply:
btw, flux actually does fetch all tags to determine the latest as well.

Reply:
Please forgive me it is all lowercase due to the smartphone.

Reply:
> they seem to be sorted by older (means the first value is always the oldest).
😄  Got it. Thanks for your explaination.